### PR TITLE
Fix splash time test

### PR DESCRIPTION
**Pull request by JuneZhou19**
@XiaowenJiang @WinnieXU1 @turtle-fly @chenge3 @xiar @fub2 
1. Menu should be on to enable splash time in test_boot.py.
2. To correct menu config option parsing. user set "on"/"off" for "menu" in config yml to enable menu, while model.py is expecting "True"/"False".

### CODE REVIEW

**Comment by WinnieXU1** [comment]
💯 🙉

**Comment by fub2** [comment]
Let's port test_esxi,py to integration test phrase.

File: infrasim/model.py [hunk 1]
```diff
@@ -1411,7 +1411,11 @@ def init(self):
         if 'boot' in self.__compute:
             self.__boot_order = self.__compute['boot'].get('boot_order', "ncd")
             if 'menu' in self.__compute['boot']:
-                self.__boot_menu = "on" if self.__compute['boot']['menu'] is True else "off"
+                menu_option = str(self.__compute['boot']['menu']).strip(" ").lower()
```
Comment (JuneZhou19) [inline infrasim/model.py]: That's right, just removed it!

**Comment by JuneZhou19** [comment]
@fub2 You are correct. Just removed test_esxi.py!

File: infrasim/model.py [hunk 2]
```diff
@@ -1411,7 +1411,11 @@ def init(self):
...
+                if menu_option != "on" and menu_option != "off":
+                    raise Exception("Illegal config option. The 'menu' must be either 'on' or 'off'.")
```
Comment (turtle-fly) [inline infrasim/model.py]: 1. Please make the verification in `precheck()`
2. At the same time, it worth a unit test on legal value
3. When user are reading this error message, they may not know this is `compute node boot`'s menu, I would suggest adding this information in exception

After that, code to get self.__boot_menu can be as precise as to get self.__boot_order, this is model's convention :ghost:

**Comment by turtle-fly** [comment]
@JuneZhou19 it's clear :eyeglasses: 
How about a precheck and unit test on this option

@WinnieXU1 please be aware of these changes and keep rebasing your log branch these days

**Comment by JuneZhou19** [comment]
@turtle-fly I can do that in precheck. About unit test, do you mean a separate PR or in the same PR?

**Comment by turtle-fly** [comment]
@JuneZhou19 better to be in same PR since it's implementation and verification of an atomic feature

File: infrasim/model.py [hunk 3]
```diff
@@ -1388,6 +1388,16 @@ def precheck(self):
             except Exception as e:
                 raise e
 
+        if 'boot' in self.__compute:
+            if 'menu' in self.__compute['boot']:
+                if isinstance(self.__compute['boot']['menu'], str):
+                    menu_option = str(self.__compute['boot']['menu']).strip(" ").lower()
+                    if menu_option not in ["on", "off"]:
+                        raise Exception("Illegal config option. The 'menu' must be either 'on' or 'off'.")
+                elif not isinstance(self.__compute['boot']['menu'], bool):
```
Comment (turtle-fly) [inline infrasim/model.py]: Considerate code :1st_place_medal:

File: test/unit/test_model.py [hunk 4]
```diff
@@ -100,6 +100,46 @@ def test_set_cpu_feature_nx(self):
...
+                "storage_backend": {},
+                "networks": {}
+            }
+
+            compute = model.CCompute(compute_info)
+            compute.init()
+            compute.precheck()
+            compute.handle_parms()
+        except ArgsNotCorrect:
+            assert True
```
Comment (turtle-fly) [inline test/unit/test_model.py]: and you will find your prompt `Illegal config option` in exception value

**Comment by turtle-fly** [comment]
@fub2 see if you can approve this PR now

**Pull request merged**